Fixed case for urls containing non-encoded symbols - rerun tests

IvanTamashevich opened the pull request:
Seems like getRequestURL() returns non-encoded URLS leading to

java.lang.IllegalArgumentException: java.net.URISyntaxException: Illegal character in path at index 45: /xxx/yyy/zzz/99.99.9999+Xxxxx+Yyy+Zzz+(Xxx+->+Zzz)
	at com.github.kristofa.brave.servlet.ServletHttpServerRequest.getUri(ServletHttpServerRequest.java:27)
	at com.atlassian.confluence.web.filter.ZipkinTracingFilter$1.lambda$create$0(ZipkinTracingFilter.java:38)

IvanTamashevich commented:
Although haven't seen a proof in the servlet spec, but according to others investigations:

https://stackoverflow.com/questions/966077/java-reading-undecoded-url-from-servlet
https://stackoverflow.com/questions/4931323/whats-the-difference-between-getrequesturi-and-getpathinfo-methods-in-httpservl/21046620#21046620

adriancole commented on archive/brave-web-servlet-filter/src/main/java/com/github/kristofa/brave/servlet/ServletHttpServerRequest.java:
```diff
@@ -24,7 +33,13 @@ public URI getUri() {
+        try
+        {
+            return URI.create(URLEncoder.encode(url.toString(), Charset.defaultCharset().name()));
+        } catch (UnsupportedEncodingException uee){
```
Change to fine level as if this ever happened it could happen a lot and fill logs

IvanTamashevich commented:
@adriancole,  shame on me, but I've missed failed httpUrlTagIncludesQueryParams test - URLEncoder is too rough approach. Will look for a more graceful one.
Let it hang there for a while if you don't mind.

adriancole commented:
:thumbsup:

IvanTamashevich commented:
@adriancole m thats it - I'm happy with the result. Although had to add Guava19 dependancy to the archive/filter branch.

adriancole commented:
sorry Ivan I can't merge this as it adds guava as a new dependency to everyone using the old version of brave. we need to find another way.

adriancole commented on archive/brave-web-servlet-filter/src/main/java/com/github/kristofa/brave/util/ModifiedUrlEncoder.java:
```diff
@@ -0,0 +1,318 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
```
sadly we can't do this as it is GPL. appreciate the thought, though.

how about looking at okhttp's HttpUrl type? you can either copy/paste from it or shade it I think

IvanTamashevich replied:
:D Ok, I'll look into it. Became too personal to let it go)

adriancole commented:
:) thanks for the tenacity!

On 30 Oct 2017 22:25, "IvanTamashevich" <notifications@github.com> wrote:

> *@IvanTamashevich* commented on this pull request.
> ------------------------------
>
> In archive/brave-web-servlet-filter/src/main/java/com/
> github/kristofa/brave/util/ModifiedUrlEncoder.java
> <https://github.com/openzipkin/brave/pull/528#discussion_r147717159>:
>
> > @@ -0,0 +1,318 @@
> +/**
> + * Modified java.net.UrlEncoder to process gracefull convercion of URLs
> + */
> +
> +/*
> + * Copyright (c) 1995, 2013, Oracle and/or its affiliates. All rights reserved.
> + * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
> + *
> + * This code is free software; you can redistribute it and/or modify it
> + * under the terms of the GNU General Public License version 2 only, as
>
> :D Ok, I'll look into it. Became too personal to let it go)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/brave/pull/528#discussion_r147717159>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAD611q18zXecnUqd8Rla5IeI8AnXapyks5sxdx1gaJpZM4QEPj_>
> .
>

IvanTamashevich commented:
Unfortunately, too many import lines to just copy okhttp util. Will drag a numerous files with it.
So working with existing implementation and switching to a "power-consuming" slow emergency one.

adriancole commented:
java 6

On Wed, Nov 1, 2017 at 3:50 AM, IvanTamashevich <notifications@github.com>
wrote:

> Seems like no fun with 1.8 streams for me.
> What is the target version for the archive project?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/brave/pull/528#issuecomment-340826919>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAD612xa8ewoZOeEvoF9Ug1m8yVgmklpks5sx0_VgaJpZM4QEPj_>
> .
>

IvanTamashevich commented:
And now need some time to get what is wrong with the test - runs fine on local env...

IvanTamashevich commented:
@adriancole I wonder if there are any issues with circleci, because I'm getting random errors (the failure cause changed when I added couple of comment lines), which are not even connected with the code I've changed and all runs OK on my local machine.

adriancole commented:
I have starred this to look at monday.. pretty beat from a tracing
workshop. Thanks!

adriancole commented on archive/brave-web-servlet-filter/src/main/java/com/github/kristofa/brave/util/UriEscaperUtil.java:
```diff
@@ -0,0 +1,56 @@
+package com.github.kristofa.brave.util;
```
move to internal package and fine by me src//main/java/com/github/kristofa/brave/servlet/internal

IvanTamashevich replied:
Done

adriancole commented:
probably just a build flake

adriancole commented:
closed via 3c27c33678a55ef1440a8e35b4b439acc411e0ad

thanks for the tenacity!
